test/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_service_test.clj

```diff
@@ -395,20 +396,25 @@
             #"shutdown-on-error triggered because of exception"
             :error))))))
 
-(deftest ^:integration overflow-test
-  (let [watch-dir (fs/temp-dir "overflow")
-        results (atom [])
-        callback (make-callback results)]
-    (with-app-with-config
-     app watch-service-and-deps {}
-     (let [service (tk-app/get-service app :FilesystemWatchService)]
-       (watch! service watch-dir callback))
-     ;; From experimenting the buffer seems to fill at 512 events.
-     ;; However the number of events varies by operating system.
-     ;; Once the buffer overflows we cannot reliably receive any other
-     ;; kind of event.
-     (dotimes [n 1000]
-       (spit (format "%s/%s.txt" watch-dir n) "file content"))
-     (let [events #{{:type :unknown
-                     :path nil}}]
-       (is (= events (wait-for-events results events)))))))
+(def overflow-event
+  (reify
+    java.nio.file.WatchEvent
+    (kind [this] java.nio.file.StandardWatchEventKinds/OVERFLOW)
+    (count [this] 1)
+    (context [this] nil)))
+
+(deftest process-events-handles-overflows-correctly
```

Comment:
Having at least one `testing` call here to describe what is going on would be good.